File: src/test/java/uk/gov/justice/digital/hmpps/oauth2server/timed/DeleteDisabledUsersTest.java
package uk.gov.justice.digital.hmpps.oauth2server.timed;

import com.microsoft.applicationinsights.TelemetryClient;
import org.junit.Before;
import org.junit.Test;
import org.junit.runner.RunWith;
import org.mockito.Mock;
import org.mockito.junit.MockitoJUnitRunner;

import static org.mockito.Mockito.verify;
import static org.mockito.Mockito.when;

@RunWith(MockitoJUnitRunner.class)
public class DeleteDisabledUsersTest {
    @Mock
    private DeleteDisabledUsersService service;
    @Mock
    private TelemetryClient telemetryClient;

    private DeleteDisabledUsers deleteDisabledUsers;

    @Before
    public void setUp() {
        deleteDisabledUsers = new DeleteDisabledUsers(service, telemetryClient);
    }

    @Test
    public void findAndDeleteDisabledUsers() {
        when(service.processInBatches()).thenReturn(0);
        deleteDisabledUsers.findAndDeleteDisabledUsers();
        verify(service).processInBatches();
    }
}

File: src/test/java/uk/gov/justice/digital/hmpps/oauth2server/timed/DeleteDisabledUsersTest.kt
package uk.gov.justice.digital.hmpps.oauth2server.timed

import com.microsoft.applicationinsights.TelemetryClient
import com.nhaarman.mockitokotlin2.mock
import com.nhaarman.mockitokotlin2.whenever
import org.junit.jupiter.api.Test
import org.mockito.Mockito.verify

class DeleteDisabledUsersTest {
  private val service: DeleteDisabledUsersService = mock()
  private val telemetryClient: TelemetryClient = mock()
  private val deleteDisabledUsers: DeleteDisabledUsers = DeleteDisabledUsers(service, telemetryClient)

  @Test
  fun findAndDeleteDisabledUsers() {
    whenever(service.processInBatches()).thenReturn(0)
    deleteDisabledUsers.findAndDeleteDisabledUsers()
    verify(service).processInBatches()
  }
}
